spk/sickchill/Makefile

```diff
@@ -16,16 +16,34 @@ DEPENDS += cross/libsodium
 ENV += SODIUM_INSTALL=system
 
 # [cryptography]
-ENV += PYO3_CROSS_LIB_DIR=$(STAGING_INSTALL_PREFIX)/lib/
-ENV += PYO3_CROSS_INCLUDE_DIR=$(STAGING_INSTALL_PREFIX)/include/
+DEPENDS += cross/cryptography
+# ENV += PYO3_CROSS_LIB_DIR=$(STAGING_INSTALL_PREFIX)/lib/
+# ENV += PYO3_CROSS_INCLUDE_DIR=$(STAGING_INSTALL_PREFIX)/include/
+# ENV += OPENSSL_LIB_DIR=$(STAGING_INSTALL_PREFIX)/lib/
+# ENV += OPENSSL_INCLUDE_DIR=$(STAGING_INSTALL_PREFIX)/include/
+
+# [greenlet]
+# Fix wheel building with older compilers
+ifeq ($(call version_lt, $(TC_GCC), 5.0),1)
+WHEELS_CPPFLAGS = [greenlet] -std=c++0x
+endif
+#ifeq ($(call version_ge, ${TCVERSION}, 7.0),1)
+#ifeq ($(call version_ge, ${TC_GCC}, 5.0),1)
+#WHEELS += src/requirements-crossenv-greenlet.txt
+#else
+#CXXFLAGS += -std=c++11
+#WHEELS += src/requirements-crossenv-greenlet-gcc4.txt
+#endif
+#DEPENDS += cross/greenlet-2.0.2
 
```

Comment:
The following works for me (in my current homeassistant branch)

NB: you need the include of spksrc.common.mk (or define the whole block after include of spksrc.spk.mk in this file) - otherwise `version_lt` is not defined and `ifeq ($(call version_lt, ${TC_GCC}, 5.0),1)` always evaluates to false

```suggestion
include ../../mk/spksrc.common.mk

# [greenlet]
ifeq ($(call version_lt, ${TC_GCC}, 5.0),1)
WHEELS_CPPFLAGS = [greenlet] -std=c++11
endif

```

so you can drop the files in `cross/greenlet-2.0.2` and the `requirements-crossenv-greenlet*` files.

Reply:
If this works out I'd recommend also updating py310-311 examples so we can later refer to them.

Reply:
Worked for all DSM6 apart from ARMv5 88f628x
```
2023-06-18T06:03:10.3128434Z       /github/workspace/toolchain/syno-88f6281-6.2.4/work/arm-marvell-linux-gnueabi/bin/arm-marvell-linux-gnueabi-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/github/workspace/toolchain/syno-88f6281-6.2.4/work/arm-marvell-linux-gnueabi/arm-marvell-linux-gnueabi/libc/usr/include -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -L /github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/lib -I /github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include -I/github/workspace/toolchain/syno-88f6281-6.2.4/work/arm-marvell-linux-gnueabi/arm-marvell-linux-gnueabi/libc/usr/include -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -L /github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/lib -I /github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include -I/github/workspace/toolchain/syno-88f6281-6.2.4/work/arm-marvell-linux-gnueabi/arm-marvell-linux-gnueabi/libc/usr/include -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include/python3.10 -I/github/workspace/toolchain/syno-88f6281-6.2.4/work/arm-marvell-linux-gnueabi/arm-marvell-linux-gnueabi/libc/usr/include -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include/python3.10 -std=c++11 -fPIC -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/crossenv/cross/include -I/github/workspace/spk/sickchill/work-88f6281-6.2.4/install/var/packages/sickchill/target/include/python3.10 -c src/greenlet/greenlet.cpp -o build/temp.linux-arm-cpython-310/src/greenlet/greenlet.o
2023-06-18T06:03:10.3132100Z       cc1plus: error: unrecognized command line option '-std=c++11'
2023-06-18T06:03:10.3132823Z       error: command '/github/workspace/toolchain/syno-88f6281-6.2.4/work/arm-marvell-linux-gnueabi/bin/arm-marvell-linux-gnueabi-gcc' failed with exit code 1
2023-06-18T06:03:10.3133282Z       [end of output]
```

Reply:
Would this work?
```
ifeq ($(call version_lt, ${TC_GCC}, 5.0),1)
ifeq ($(call version_lt, ${TC_GCC}, 4.8),1)
WHEELS_CPPFLAGS = [greenlet] -std=C++0x
else
WHEELS_CPPFLAGS = [greenlet] -std=c++11
endif
endif
```

Reply:
> -std=C++0x

No, it fails with
```
cc1plus: error: unrecognized command line option '-std=C++0x'
```

Reply:
oops, there was a typo, it should be `-std=c++0x` with lowercase 'c'
but it fails too, first error is
```
src/greenlet/greenlet.cpp:709:8: error: 'thread_local' does not name a type
```